include/zephyr/fs/fs.h

```diff
@@ -262,8 +262,8 @@ static inline void fs_dir_t_init(struct fs_dir_t *zdp)
  *   - @c FS_O_CREATE create file if it does not exist
  *   - @c FS_O_APPEND move to end of file before each write
  *
- * If @p flags are set to 0 the function will attempt to open an existing file
- * with no read/write access; this may be used to e.g. check if the file exists.
+ * @warning If @p flags are set to 0 the function will attempt to open file with
+ *          no read/write access.
```

Comment:
Why does this have a @warning ? It seems to be an allowed usage pattern but why does it have a warning,

Reply:
Because you will open file but will not be able to read/write to it.
Maybe. Because backends may interpret some of open flags differently, for example create may imply write on one backend but not another.

Reply:
The problem I have with the `@warning` is that (without some extra information) the user does not know what is warned against.

Reply:
I don't know, seems that the info no lack of read/write access is enough.

Reply:
I agree, but when adding the `@warning` it should warn against something. Either remove the warning or add something like: `When issued on a existing file the file needs to be closed afterwards. When issued on a non-existing file an error message will be generated`

Reply:
I do not want to mention error message, because any subsystem called by driver may actually render error messages, or not depending on Kconfig, so it is pointless to provide such message here.